src/main/java/impl/image/ConvolutionProviderFactory.java

```diff
@@ -2,8 +2,41 @@
 
 import api.image.ConvolutionProvider;
 
+import java.awt.*;
+import java.lang.Math.*;
 public class ConvolutionProviderFactory {
-    public static ConvolutionProvider getInstance() {
-        return null;
+    private static ConvolutionProviderFactory instance;
+    public static ConvolutionProviderFactory getInstance()
+    {
+        if (instance == null)instance = new ConvolutionProviderFactory();
```

Comment:
You should create the new class which implements `ConvolutionProvider` interface and return it's object here.  
Method signature should stay `public static ConvolutionProvider getInstance()`